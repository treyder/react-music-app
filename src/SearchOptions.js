import React from "react";
import Button from 'material-ui/Button';

class SearchOptions extends React.PureComponent {

    constructor(props) {
        super(props);
        this.state = {
            titleOptionColor: 'primary',
            genreOptionColor: 'secondary'
        };
        this.handleTitleOptionChange = this.handleTitleOptionChange.bind(this);
        this.handleGenreOptionChange = this.handleGenreOptionChange.bind(this);
    }

    handleTitleOptionChange() {
        this.setState({
            titleOptionColor: 'primary',
            genreOptionColor: 'secondary'
        });
    }

    handleGenreOptionChange() {
        this.setState({
            titleOptionColor: 'secondary',
            genreOptionColor: 'primary'
        });
    }

    render() {
        return (
            <div className="Search-options-table">
                <div className="Search-options-row">
                    <div className="Search-options-cell">SEARCH BY</div>
                    <Button size="small" id="titleButton" variant="raised" color={this.state.titleOptionColor} onClick={this.handleTitleOptionChange}>
                        TITLE
                    </Button>
                    <Button size="small" id="genreButton" variant="raised" color={this.state.genreOptionColor} onClick={this.handleGenreOptionChange}>
                        GENRE
                    </Button>
                </div>
            </div>
        );
    }
}

export default SearchOptions;