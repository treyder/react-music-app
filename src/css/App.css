.App {
  text-align: center;
  height: 100%;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.Search-container {
  background-color: #222;
  padding: 20px;
  color: white;
  text-align: center;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.Search-header {
  font-size: large;
  padding: 20px;
}

.Search-input {
  min-width: 500px;
}

.Search-option-selected {
  background-color: red;
}

.Search-option-default {
  background-color: grey;
}

.Search-options-table {
  display: table;
}

.Search-options-row {
  width: 100%;
  display: table-row;
}

.Search-options-cell {
  border: 0px;
	display: table-cell;
	padding: 3px 10px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.Search-result {
  background-color: black;
  color: white;
}

.Footer {
  background-color: black;
  color: red;
  text-align: left;
  padding: 30px;
}
