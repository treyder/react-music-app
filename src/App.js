import React, { Component } from 'react';
import logo from './logo.svg';
import AppTitle from './AppTitle';
import SearchHeader from './SearchHeader';
import SearchForm from './SearchForm';
import ErrorBoundary from './ErrorBoundary';
import Footer from './Footer';

import './css/App.css';

class App extends Component {
  render() {
    return (
      <ErrorBoundary>
        <div className="App">

          <div className="Search-container">
            <img src={logo} className="App-logo" alt="logo" />
            {AppTitle}
            {SearchHeader}
            <SearchForm/>
          </div>

          {Footer}
        </div>
      </ErrorBoundary>
    );
  }
}

export default App;
