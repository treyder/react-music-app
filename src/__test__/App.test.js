import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../App';
import SearchOptions from '../SearchOptions';

import "isomorphic-fetch";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('search options renders correctly', () => {
  const tree = renderer.create(
    <SearchOptions />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

