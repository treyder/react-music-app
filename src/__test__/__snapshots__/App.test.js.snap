// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search options renders correctly 1`] = `
<div
  className="Search-options-table"
>
  <div
    className="Search-options-row"
  >
    <div
      className="Search-options-cell"
    >
      SEARCH BY
    </div>
    <button
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30 MuiButton-raisedPrimary-31 MuiButton-sizeSmall-37"
      disabled={false}
      id="titleButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-26"
      >
        TITLE
      </span>
      <span
        className="MuiTouchRipple-root-42"
      />
    </button>
    <button
      className="MuiButtonBase-root-40 MuiButton-root-25 MuiButton-raised-30 MuiButton-raisedSecondary-32 MuiButton-sizeSmall-37"
      disabled={false}
      id="genreButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-26"
      >
        GENRE
      </span>
      <span
        className="MuiTouchRipple-root-42"
      />
    </button>
  </div>
</div>
`;
